specs/agents/tracing-instrumentation-aws.md

```diff
@@ -0,0 +1,53 @@
+## AWS services spans
+
+We describe how to instrument some of AWS' services in this document.
+Many of the services have a fairly small API beyond the basic administrative commands. We therefore
+only instrument the most interesting and often-repeated operations.
+
+### S3 (Simple Storage Service)
+
+AWS Simple Storage Service offers object storage via a REST API. The objects are organized into buckets, which are 
+themselves organized into regions.
+
+- `span.name`: The span name should follow this pattern: `S3 <OPERATION_NAME> <BUCKET-NAME>`. For example,
+`S3 GET_OBJECT myBucket`. Note that the operation name is in all caps with underscores (snake case in caps).
```

Comment:
On the operation name: Any reason not to use the "GetObject"-style naming that AWS uses? E.g., https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_GetBucket.html

Reply:
Seems reasonable. I definitely prefer `snake_case`, but that's because I'm a python dev. ;) Here's a link to `GetObject`, for example: https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html

Reply:
Also FWIW, the "GetObject"-style naming are the strings used in IAM policy language. E.g. "s3:GetObject" at https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_action.html

Reply:
I chose to use the snake case in caps to be as close to the messaging spec as possible. (which uses operation names like `RECEIVE` as in `RabbitMQ RECEIVE from MyQueue`).

But @trentm, you bring up a good point regarding the IAM policy language. We do have a field `span.action` where we could use the formatting `s3:GetObject` so that the span name remains close to the messaging spec. Currently this spec dictates that, that field should be `get_object`.

If we did use the camel case for the `span.name`, would you imagine it would look like: `S3 GetObject myBucket`?

Reply:
I'm leaning towards an option you didn't list:

`span.name`: `S3 GetObject myBucket`
`span.action`: `GetObject`

I'm not very concerned with exactly matching the messaging spec -- I think it's probably more important to match S3 norms.

I don't like mixing the two forms, though. We should either go all-in on snake case or all-in on AWS names.

Reply:
What @basepi said. I think the "GetObject" form will most quickly be self-explanatory to AWS users.

Reply:
My vote would be for `span.name`: `S3 GetObject myBucket`. 

`span.action`: `GetObject` seems appropriate to align with the AWS API, unless there's a stronger preference to be consistent on the agent side, with snake casing.

Reply:
Great, thanks for your votes. We'll have it be:

```
span.name: S3 GetObject myBucket
span.action: GetObject
```